components/formats-bsd/src/loci/formats/services/EXIFServiceImpl.java

```diff
@@ -82,26 +79,48 @@ public void initialize(String file) throws ServiceException, IOException {
   @Override
   public HashMap<String, String> getTags() {
     HashMap<String, String> tagMap = new HashMap<String, String>();
-    if (directory != null) {
-      for (Tag tag : directory.getTags()) {
-        tagMap.put(tag.getTagName(), tag.getDescription());
+    if (directories != null) {
+      for (Directory directory : directories) {
```

Comment:
Taking another look at the EXIF spec, I realize this might not achieve what we want. The `Image` and the `Thumbnail` EXIF tags are located in different IFDs but the tags overlap. In the case of  `XResolution`  this means the tag will be overwritten by the ThumbnailXResolution if present.